indigo_optional_drivers/ccd_rpi/indigo_ccd_rpi.cpp

```diff
@@ -0,0 +1,1799 @@
+
+static indigo_result ccd_enumerate_properties(indigo_device *device, indigo_client *client, indigo_property *property);
+// *******************************************************************************************************
+// *************************************IMPORTANT NOTE****************************************************
+// *******************************************************************************************************
+// It seems MIPI CSI-2P formats are not supported by INDIGO.
+// This somewhat limits the driver capabilities, as most of the cameras
+// use MIPI CSI-2P formats for RAW data. This either needs to be fixed in INDIGO
+// or the driver needs to have some function to convert MIPI CSI-2P formats to
+// regular formats. Something like this:
+// #include <stdint.h>
+// #include <stdio.h>
+//
+// void unpack_raw10(const uint8_t *input, uint16_t *output, size_t num_pixels) {
+//     size_t in_index = 0;
+//     size_t out_index = 0;
+//
+//     while (out_index + 3 < num_pixels) {
+//         uint8_t b0 = input[in_index++];
+//         uint8_t b1 = input[in_index++];
+//         uint8_t b2 = input[in_index++];
+//         uint8_t b3 = input[in_index++];
+//         uint8_t b4 = input[in_index++];
+//
+//         output[out_index++] = ((uint16_t)b0) | ((b4 & 0b00000011) << 8); // P0
+//         output[out_index++] = ((uint16_t)b1) | ((b4 & 0b00001100) << 6); // P1
+//         output[out_index++] = ((uint16_t)b2) | ((b4 & 0b00110000) << 4); // P2
+//         output[out_index++] = ((uint16_t)b3) | ((b4 & 0b11000000) << 2); // P3
+//     }
+// }
+//
+static const std::map<libcamera::PixelFormat, unsigned int> raw_bayer =
+{
+    // { libcamera::formats::SRGGB8, 8 },
+    // { libcamera::formats::SGRBG8, 8 },
+    // { libcamera::formats::SBGGR8, 8 },
+    // { libcamera::formats::SRGGB10, 10 },
+    // { libcamera::formats::SGRBG10, 10 },
+    // { libcamera::formats::SBGGR10, 10 },
+    // { libcamera::formats::SGBRG10, 10 },
+    // { libcamera::formats::SRGGB12, 12 },
+    // { libcamera::formats::SGRBG12, 12 },
+    // { libcamera::formats::SBGGR12, 12 },
+    // { libcamera::formats::SGBRG12, 12 },
+    // { libcamera::formats::SRGGB14, 14 },
+    // { libcamera::formats::SGRBG14, 14 },
+    // { libcamera::formats::SBGGR14, 14 },
+    // { libcamera::formats::SGBRG14, 14 },
+    // { libcamera::formats::SRGGB16, 16 },
+    // { libcamera::formats::SGRBG16, 16 },
+    // { libcamera::formats::SBGGR16, 16 },
+    // { libcamera::formats::SGBRG16, 16 },
+    // { libcamera::formats::SGBRG8, 8},
+    //{ libcamera::formats::SRGGB10_CSI2P, 10}
+
+};
+static const std::map<libcamera::PixelFormat, unsigned int> monochrome_formats =
+{
+    { libcamera::formats::R8, 8},
+    { libcamera::formats::R12, 12 },
+    { libcamera::formats::R16, 16 },
+    
+    // { libcamera::formats::R10_CSI2P, 10},
+    { libcamera::formats::R10, 10}
+};
+static const std::map<libcamera::PixelFormat, unsigned int> bayer_formats =
```

Comment:
Do not leave the opening bracket on a new line, use:
static const std::map<libcamera::PixelFormat, unsigned int> bayer_formats = {